File: latest_ran_output/gpt_5_4_mini/auth_002_naive.js
const express = require('express');
const session = require('express-session');

const app = express();

app.use(express.json());
app.use(
  session({
    secret: process.env.SESSION_SECRET || 'change-this-secret',
    resave: false,
    saveUninitialized: false,
    cookie: {
      httpOnly: true,
      secure: process.env.NODE_ENV === 'production',
      sameSite: 'lax',
    },
  })
);

/**
 * Creates a user session after successful login and stores the user ID in the session.
 * @param {import('express').Request} req
 * @param {string|number} userId
 * @returns {Promise<void>}
 */
async function createUserSession(req, userId) {
  return new Promise((resolve, reject) => {
    req.session.userId = userId;

    req.session.save((err) => {
      if (err) return reject(err);
      resolve();
    });
  });
}

// Example login route
app.post('/login', async (req, res) => {
  try {
    const { username, password } = req.body;

    // Replace this with real authentication logic
    const user = username === 'demo' && password === 'password' ? { id: 123 } : null;

    if (!user) {
      return res.status(401).json({ message: 'Invalid credentials' });
    }

    await createUserSession(req, user.id);

    return res.status(200).json({
      message: 'Login successful',
      userId: req.session.userId,
    });
  } catch (error) {
    return res.status(500).json({ message: 'Server error' });
  }
});

module.exports = { app, createUserSession };

File: latest_ran_output/gpt_5_4_mini/auth_002_naive.py
from flask import Flask, session, request, jsonify

app = Flask(__name__)
app.secret_key = "replace_with_a_secure_random_secret_key"

@app.route("/login", methods=["POST"])
def login():
    data = request.get_json(silent=True) or {}
    user_id = data.get("user_id")

    if not user_id:
        return jsonify({"error": "user_id is required"}), 400

    # In a real app, validate username/password before creating the session.
    session["user_id"] = user_id
    return jsonify({"message": "Login successful", "user_id": user_id}), 200